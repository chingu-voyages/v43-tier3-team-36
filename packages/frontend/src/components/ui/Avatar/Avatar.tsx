/* eslint-disable react/jsx-props-no-spreading */
import type { RefAttributes, ForwardRefExoticComponent } from 'react';
import { forwardRef, useState } from 'react';
import clsx from 'clsx';
import stc from 'string-to-color';
import Image, { type ImageProps } from 'next/image';

export interface AvatarProps extends ImageProps {
  /**
   * Size of the component (width/height)
   */
  size?: number;
}

const styles = {
  base: 'rounded-full',
  image: 'object-cover rounded-[inherit]',
  fallback: 'text-black flex justify-center items-center rounded-[inherit]',
};

export const Avatar: ForwardRefExoticComponent<
AvatarProps & RefAttributes<HTMLImageElement>
> = forwardRef(({
  size = 40, alt, className, ...rest
}, ref) => {
  const [isLoadError, setIsLoadError] = useState(false);

  const containerClassNames = clsx(styles.base, className);

  return (
    <div className={containerClassNames}>
      <Image
        ref={ref}
        width={size}
        height={size}
        alt={alt}
        className={isLoadError ? styles.fallback : styles.image}
        style={{ backgroundColor: isLoadError ? stc(alt) : 'initial' }}
        onError={() => setIsLoadError(true)}
        {...rest}
      />
    </div>
  );
});

if (process.env.NODE_ENV !== 'production') {
  Avatar.displayName = 'Avatar';
}
