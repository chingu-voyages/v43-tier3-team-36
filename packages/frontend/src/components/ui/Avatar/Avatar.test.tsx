import { render } from 'test-utils';

import { Avatar } from './Avatar';

describe('<Avatar />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Avatar src="/" alt="Foo" />);
      expect(container).toMatchSnapshot();
    });
  });
});
