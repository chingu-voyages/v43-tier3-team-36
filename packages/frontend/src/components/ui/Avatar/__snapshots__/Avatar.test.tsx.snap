// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> Snapshots should match snapshot 1`] = `
<div>
  <div
    class="max-w-full rounded-full"
  >
    <img
      alt="Foo"
      class="object-cover"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="http://localhost/_next/image?url=%2F&w=96&q=75"
      srcset="/_next/image?url=%2F&w=48&q=75 1x, /_next/image?url=%2F&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </div>
</div>
`;
